src/views/docs-view/server.ts

```diff
@@ -387,15 +387,11 @@ export function getStaticGenerationFunctions<
 			/**
 			 * Filter versions to include only those where this document exists
 			 */
-			// Construct a document path that the content API will recognize
-			const pathWithoutVersion = pathParts
-				.filter((part) => part !== versionPathPart)
-				.join('/')
-			const fullPath = `doc#${path.join(basePathForLoader, pathWithoutVersion)}`
-			// Filter for valid versions, fetching from the content API under the hood
-			const validVersions = await getValidVersions(
+			const validVersions = await fetchValidVersions(
```

Comment:
Nice improvement. I think the risk with these types of sprawling function is to add yet another line of transforms so moving those transforms into `getValidVersions` is move towards better encapsulation.